feat: Theme Hacker Owl

koolamusic opened the pull request:
Hi Guys, created something here out of a little burning need, I'd be using remix a lot and thus wanted an IDE with a bit of a modern look and feel. 

I consider this continuous WIP, and would take responsibility for this theme even as I continue to use it for my work. 
Adds a monospace font and sans-serif and also, I'd find a way to map more `ace-identifier` keywords to the syntax highlight config. 

Also I create theme color variables using CSS Variables, so that the theme can easily be adapted. 
Feedback is welcome, like I said, this will be a continuous effort. 

Thanks


https://user-images.githubusercontent.com/8960757/131011195-957e65ba-0b99-46a3-b28d-fc6da2cc7afc.mov

LianaHus commented on apps/remix-ide/src/app/tabs/theme-module.js:
```diff
@@ -7,6 +7,7 @@ const _paq = window._paq = window._paq || []
 
 const themes = [
   { name: 'Dark', quality: 'dark', url: 'assets/css/themes/remix-dark_tvx1s2.css' },
+  { name: 'HackerOwl', quality: 'dark', url: 'assets/css/themes/remix-night_owl.css' },
```
Please put it after the Light theme

koolamusic replied:
sure. will push an update

LianaHus commented on apps/remix-ide/src/app/tabs/theme-module.js:
```diff
@@ -8,10 +8,10 @@ const _paq = window._paq = window._paq || []
   { name: 'Midcentury', quality: 'light', url: 'assets/css/themes/remix-midcentury_hrzph3.css' },
   { name: 'Black', quality: 'dark', url: 'assets/css/themes/remix-black_undtds.css' },
   { name: 'Candy', quality: 'light', url: 'assets/css/themes/remix-candy_ikhg4m.css' },
-
```
the line is separating custom-made themes from ready ones but maybe it is fine to remove it

LianaHus commented:
> @LianaHus had time to review?

looks good to me. thanks for your contribution! :)

LianaHus reviewed (changes requested):
since there is no borders for sidepanel we need different colors for the background

koolamusic commented:
> since there is no borders for sidepanel we need different colors for the background
> 
> ![Screenshot from 2021-09-06 11-19-20](https://user-images.githubusercontent.com/9685356/132193174-14c0ee20-e652-4f27-bc16-418e20e787a1.png)
> ![Screenshot from 2021-09-06 11-19-37](https://user-images.githubusercontent.com/9685356/132193201-8dd04941-a1eb-4172-a635-fb30cb2c2736.png)
> ![Screenshot from 2021-09-06 11-20-04](https://user-images.githubusercontent.com/9685356/132193217-49ad398f-cd63-434b-9e52-abf7eae49ba6.png)

Noted

koolamusic commented:
![Screenshot 2021-09-06 at 11-21-12 Remix - Ethereum IDE](https://user-images.githubusercontent.com/8960757/132210772-32f3f05e-0530-40b7-956e-5021e737ca38.png)


@LianaHus added the patches.

LianaHus commented:
> ![Screenshot 2021-09-06 at 11-21-12 Remix - Ethereum IDE](https://user-images.githubusercontent.com/8960757/132210772-32f3f05e-0530-40b7-956e-5021e737ca38.png)
> 
> @LianaHus added the patches.

i still don't see the 'border' between switch tab and terminal

koolamusic commented:
> > ![Screenshot 2021-09-06 at 11-21-12 Remix - Ethereum IDE](https://user-images.githubusercontent.com/8960757/132210772-32f3f05e-0530-40b7-956e-5021e737ca38.png)
> > @LianaHus added the patches.
> 
> i still don't see the 'border' between switch tab and terminal

I feel like I am missing something. 
is there a way you could annotate the section.

koolamusic commented:
![image](https://user-images.githubusercontent.com/8960757/133075608-1ed3f6b8-7620-4ecf-a282-a0d8359400fb.png)


@LianaHus see this

LianaHus commented:
the contrast between right border of panel and bg is not enough to see it but it is visible anyway, so if you like it like that I don't mind

LianaHus reviewed (changes requested):
I just realize you are using extra classes only for this them, can we avoid this?

koolamusic commented:
ok, let me take a look at it

koolamusic commented:
<img width="1677" alt="image" src="https://user-images.githubusercontent.com/8960757/136012454-87759b7b-5252-4a50-8dcd-1619034d3d92.png">


Added the fixes and changed the sidebar panel color to fix concerns with the contrast between the left and right panel.

koolamusic commented:
@LianaHus can we please go over this again, what do I need to move this PR ?

Hbutlercapone commented:
I'll merge all tonight I apoligize I had to move into a new home

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: Rob ***@***.***>
Sent: Tuesday, July 5, 2022 12:42:54 PM
To: ethereum/remix-project ***@***.***>
Cc: Heather ***@***.***>; Comment ***@***.***>
Subject: Re: [ethereum/remix-project] feat: Theme Hacker Owl (#1527)


Our back and forth has taken pretty long time. We'd like to either merge this or close your pr soon. If you can make these updates, we can get this in a release very soon). The next release is in 2 weeks.

  *    In the settings Hacker Owl should go after Candy. ( we do this)
  *    the background of the input fields is too light causing the place holder text need a bit more contrast to make it more legible.
  *    new variables in the editor are not defined in your theme:

[image]<https://user-images.githubusercontent.com/6050001/177374805-13c5f6ec-c4ec-4168-907f-ba44e4bec2ea.png>

here there slightly visible:

[image]<https://user-images.githubusercontent.com/6050001/177374870-4214602e-c6f9-4468-aae4-43524083b725.png>

  *    remove contextView.js (right click in editor). Please rebase. You probably had put this in place before we switched to Monaco API.

—
Reply to this email directly, view it on GitHub<https://github.com/ethereum/remix-project/pull/1527#issuecomment-1175267015>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AT5C7EDTBYPSY3IEDRRK2B3VSRQY5ANCNFSM5C33XS7A>.
You are receiving this because you commented.Message ID: ***@***.***>

koolamusic commented:
@Hbutlercapone do you need any assistance with this?

Hbutlercapone reviewed (approved):
Approve merge of 24 commits for feat:Theme Hacker Owl #1527

koolamusic commented:
> 

these issues have been resolved

Hbutlercapone commented:
Its saying its pending two other reviewers reviews

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: U.M Andrew ***@***.***>
Sent: Friday, July 29, 2022 8:13:22 AM
To: ethereum/remix-project ***@***.***>
Cc: Heather ***@***.***>; Mention ***@***.***>
Subject: Re: [ethereum/remix-project] feat: Theme Hacker Owl (#1527)


Our back and forth has taken pretty long time. We'd like to either merge this or close your pr soon. If you can make these updates, we can get this in a release very soon). The next release is in 2 weeks.

* [ ]  In the settings Hacker Owl should go after Candy.  ( we do this)

* [ ]  the background of the input fields is too light causing the place holder text need a bit more contrast to make it more legible.

* [ ]  new variables in the editor are not defined in your theme:


[image]<https://user-images.githubusercontent.com/6050001/177374805-13c5f6ec-c4ec-4168-907f-ba44e4bec2ea.png> here there slightly visible: [image] <https://user-images.githubusercontent.com/6050001/177374870-4214602e-c6f9-4468-aae4-43524083b725.png>

* [ ]  remove contextView.js  (right click in editor).  Please rebase. You probably had put this in place before we switched to Monaco API.


[image]<https://user-images.githubusercontent.com/8960757/181755792-83570653-c08c-46a4-b8d4-902a29f33c0b.png>

[image]<https://user-images.githubusercontent.com/8960757/181755931-955e2e78-9bd4-460f-8dcd-6760f98f8356.png>

These issues have been resolved

—
Reply to this email directly, view it on GitHub<https://github.com/ethereum/remix-project/pull/1527#issuecomment-1199206042>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AT5C7EFTL5W3SKXGJLTRGQLVWPDGFANCNFSM5C33XS7A>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

koolamusic commented:
@ryestew <> @LianaHus  ^^

ryestew commented on apps/remix-ide/src/app/editor/contextView.js:
```diff
@@ -0,0 +1,194 @@
+      this._nodes = nodes
+      this.update()
+    })
+    this.contextualListener.event.register('stopHighlighting', () => {
```
this line is triggering a linting error: Unexpected empty arrow function

ryestew reviewed (changes requested):
Thanks so much for your update.  There is a linting error that was triggered which needs to be fixed.  We might have another comment in a day or two.  Almost there!

ryestew reviewed (changes requested):
Please remove all changes to contextView.js and we will take care of that part. So the only files that should be in your pr are remix-hacker_owl.css and theme-module.js.  Then we can merge it!  Thanks

Aniket-Engg commented:
> Please remove all changes to contextView.js and we will take care of that part. So the only files that should be in your pr are remix-hacker_owl.css and theme-module.js. Then we can merge it! Thanks

@koolamusic Can you consider this?

koolamusic commented:
> > Please remove all changes to contextView.js and we will take care of that part. So the only files that should be in your pr are remix-hacker_owl.css and theme-module.js. Then we can merge it! Thanks
> 
> @koolamusic Can you consider this?

Yes. on it

ryestew reviewed (approved):
Thanks and thanks for your patience @koolamusic!!

Aniket-Engg commented:
Thanks for your continuous co-operation and this great contribution @koolamusic !!!

koolamusic commented:
My pleasure ❤️
